dispatcher/backend/src/tests/integration/routes/workers/test_worker.py

```diff
@@ -212,32 +213,130 @@ def test_checkin_another_user(
 
 
 class TestWorkerRequestedTasks:
-    def test_requested_task_worker_as_admin(self, client, access_token, worker):
+    @pytest.fixture()
+    def req_task_query_string(self, worker):
+        return {
+            "worker": worker["name"],
+            "avail_cpu": 4,
+            "avail_memory": 2048,
+            "avail_disk": 4096,
+        }
+
+    @pytest.fixture
+    def make_headers(self):
+        def _make_headers(access_token: str, client_ip: str) -> Dict[str, Any]:
+            return {
+                "Authorization": access_token,
+                "X-Forwarded-For": client_ip,
+            }
+
+        return _make_headers
+
+    @pytest.fixture
+    def admin_headers(self, make_headers, access_token, default_ip):
+        def _admin_headers(
+            access_token: str = access_token, client_ip: str = default_ip
+        ) -> Dict[str, Any]:
+            return make_headers(access_token=access_token, client_ip=client_ip)
+
+        return _admin_headers
+
+    @pytest.fixture
+    def worker_headers(self, make_headers, make_access_token, worker, default_ip):
+        def _worker_headers(
+            access_token: str = make_access_token(worker["username"], "worker"),
+            client_ip: str = default_ip,
+        ) -> Dict[str, Any]:
+            return make_headers(access_token=access_token, client_ip=client_ip)
+
+        return _worker_headers
+
+    @pytest.fixture
+    def default_ip(self):
+        return "192.168.1.1"
+
+    @pytest.fixture
+    def increase_ip(self):
```

Comment:
Why not using functions for fixture? Is it to keep this together with the tests that use them? Maybe it's time to split the file 😉

Reply:
Thank you for confirming the idea of splitting the file was a good one 🤣